### PR TITLE
React capstone project

### PR DESCRIPTION
**Pull request by Elijahdre**
# Crypto metrics webapp

- [ ] The project is a single-page application (SPA) built with React and Redux.
- [ ] The data retrieved from the API should be stored in the Redux store.
- [ ] You should filter the data that you retrieve from the API using a Filter stateless component.
- [ ] Every page (the main page and the pages for each item) should have a unique route within the SPA.
- [ ] The project should be deployed and accessible online (add a link to the online version of your app to your README file).
- [ ] link to [video](https://www.loom.com/share/87092be77bf942c795edf7436ef9f588)

### CODE REVIEW

File: src/components/Homepage.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { fetchCrypto } from '../redux/crypto/crypto';
+import Currencies from './Currencies';
+
+function Homepage() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchCrypto());
+  }, []);
+
```
Comment (thecodechaser) [inline src/components/Homepage.js:13]: - [ ] You did really a great job on this project but you missed a major requirement. Please add a feature on the home page to filter cryptos. (You should include a search feature on the page).
image

File: README.md [hunk 2]
```diff
@@ -1,75 +1,125 @@
...
+- [x] Handle events in a React app.
+- [x] Write integration tests with a React testing library.
+- [x] Use styles in a React app.
+- [x] Use React life cycle methods.
+- [x] Apply React best practices and language style guides in code.
+- [x] Use store, actions and reducers in React.
 
-## Live Demo (if available)
+## Project Documentation 📄
+- [x] Here is the presentation video [link]()👈
```
Comment (thecodechaser) [inline README.md]: - [ ] Please make sure to show(highlight) some interesting piece of code or something you built that you are very proud of in the presentation video as requested in the project requirements. (You may want to record the video again with highlighted code and also fix the video link in the README file).

File: README.md [hunk 3]
```diff
@@ -1,75 +1,125 @@
...
+- [x] Front-end library: React with Redux Toolkit
+- [x] API:
+- [x] Code Editor: VS Code
 
-## Getting Started
+## Website Mockup 📱 💻 🖥️
+![app_screenshot]()
+
+## Screenshots 📸
+Screenshots can be found in the repository.
```
Comment (thecodechaser) [inline README.md]: - [ ] The mockup and screenshots are broken in the README file, Please make sure to fix this issue.

File: README.md [hunk 4]
```diff
@@ -1,75 +1,125 @@
...
 
-**This is an example of how you may give instructions on setting up your project locally.**
-**Modify this file to match your project, remove sections that don't apply. For example: delete the testing section if the currect project doesn't require testing.**
+## Live Demo 🔗
 
+Check the live demo [netlify]()👈
+
```
Comment (thecodechaser) [inline README.md]: - [ ] The live version of the application is broken in the README file, Please make sure the project should be deployed and accessible online also add deployed link in the README file.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Elijahdre👋, 

### You are doing a marvelous job on the React capstone project and this is one of the best projects I have seen so far!

There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Homepage.js (hunk 1 above)
Comment (Elijahdre) [reply src/components/Homepage.js:13]: @thecodechaser  Thank you for the review. This is just a project description. If you can check the technical requirement which I paid the most attention to, that feature was not included also on the project screenshot.

File: src/components/Homepage.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <div>
+      <Currencies />
+    </div>
+  );
+}
```
Comment (tresorsawasawa) [inline src/components/Homepage.js:19]: - [ ] As mentioned by the previous reviewer and for the project requirements, kindly make sure your **Home Page** has a list of categories that could be filtered using given parameters, **e.g. category name**. I suggest you add a search bar so you can type the name of the crypto you need and get it from the list.

File: src/components/Homepage.js (hunk 1 above)
Comment (tresorsawasawa) [reply src/components/Homepage.js:13]: Hi @Elijahdre. I think if you correctly check the requirements for the Home page you will find that there is a comment telling what the review said.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @Elijahdre 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Homepage.js (hunk 1 above)
Comment (tresorsawasawa) [reply src/components/Homepage.js:13]: Hi @Elijahdre. I think if you correctly check the requirements for the Home page you will find that there is a comment telling what the reviewer said.

File: src/components/Homepage.js (hunk 5 above)
Comment (Elijahdre) [reply src/components/Homepage.js:19]: @tresorsawasawa 
cr
Thank you but I don't see a search option on the technical requirements. Many other projects have been approved and they didn't have that too. My partner's was approved.

Comment (tresorsawasawa) [reply src/components/Homepage.js:19]: Hi @Elijahdre 

Adding the **search option** is a suggestion to help you match the requirement saying that: **"The Home Page has a list of categories that could be filtered using given parameters, e.g. category name"**
Screenshot from 2022-10-08 02-53-02

I don't know if it is clear

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @Elijahdre, 👋🏻👋🏻 

Great work on making the changes requested by a previous reviewer 👏 💪🏻💪🏻🔥🔥
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights ✨✨
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the functionality and design✅
📌 Good commit messages ✅
📌 Professional README ✅

 ## Required Changes ♻️
_Check the comments under the review._
- [ ] As mentioned by the previous reviewer and for the project requirements, kindly make sure your Home Page has a list of categories that could be filtered using given parameters, e.g. category name. I suggest you add a search bar so you can type the name of the crypto you need and get it from the list.

I think it is one of the technical requirements since you didn't use the filter so the search bar is one way but you can make it in anyway you like  💪🏻💪🏻🔥🔥
image

> Adding the **search option** is a suggestion to help you match the requirement saying that: **"The Home Page has a list of categories that could be filtered using given parameters, e.g. category name"** Screenshot from 2022-10-08 02-53-02
> 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by medaminedev66 — APPROVED** [review]
Hi @Elijahdre 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- Your project works as expected and runs without errors.
- The feature of filtering the data has been implemented.
- Your repo looks professional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Elijahdre** [comment]
@medaminedev66 Thank you for the review.

File: src/components/Homepage.js (hunk 5 above)
Comment (Elijahdre) [reply src/components/Homepage.js:19]: @tresorsawasawa Yes it is clear. Thank you

**Pull request merged**